core/src/main/java/com/linecorp/armeria/client/endpoint/OrElseEndpointGroup.java

```diff
@@ -18,25 +18,27 @@
 
 import static java.util.Objects.requireNonNull;
 
-import java.util.List;
+import java.util.Collection;
 
 import com.linecorp.armeria.client.Endpoint;
 
-final class OrElseEndpointGroup implements EndpointGroup {
+final class OrElseEndpointGroup extends EndpointGroup {
     private final EndpointGroup first;
     private final EndpointGroup second;
 
     OrElseEndpointGroup(EndpointGroup first, EndpointGroup second) {
         this.first = requireNonNull(first, "first");
         this.second = requireNonNull(second, "second");
+        updateEndpoints(null);
```

Comment:
We don't really need to add the Listenable interface indeed, but I'm more concerned with the simple or-else behavior is now implemented with all the complexity, including locks and such of the delegates it's copying data from. If we want to make the or-else Listenable, we should only do that rather than making it a fully managed endpoint group I think. Instead of a separate interface we could also just move the listener related methods into the simple EndpointGroup interfaces. What do you think?